coding-articles/natachi-article-introduction-smart-contract-auditing/Article.md

```diff
@@ -0,0 +1,494 @@
+    
+
+That's it! You're now ready to start using Foundry. If you require additional information or assistance, please visit the Foundry [repository](https://github.com/foundry-rs/foundry#installation) or check out the [Foundry book](https://book.getfoundry.sh/getting-started/installation).
+
+### Auditing Smart Contracts
+
+Let's review the steps in the auditing process before we start. The process of auditing a smart contract typically includes the following steps:
+
+1. Running tests - This step involves running various tests on the smart contract code to identify any potential issues.
+    
+2. Reading specifications and documentation - This step involves thoroughly reading the specifications and documentation for the smart contract to gain a deep understanding of how it works.
+    
+3. Running fast tools - Tools like Slither are used to quickly identify potential vulnerabilities and security issues in the code.
+    
+4. Manual analysis - After running the tests and using tools, a manual analysis of the code is performed to identify any issues that may have been missed by the automated processes.
+    
+5. Discussion - Any identified issues are discussed to ensure a thorough understanding of the problem and to determine the best course of action.
+    
+6. Writing the report - Finally, a comprehensive report is written that documents all the identified issues, including recommendations for fixing any problems. The report can be used as a reference for future updates or audits of the smart contract.
+    
+
+Okayyy! Let's jump into auditing a code base!
+
+Remember, we'll be working on the project using Foundry and Slither. [Click here](https://github.com/natachigram/audit-practice.git) to access the project's github repository.
+
+1. **First, open Vscode. Then, make a new folder and clone the** [**github repo**](https://github.com/natachigram/audit-practice.git) **for this task. After that, go to your terminal and type "forge install".**
+    
+    ```plaintext
+    $ https://github.com/natachigram/audit-practice.git
+    $ cd audit-practice
+    $ forge install
+    ```
+    
+    In the <mark>audit/src/ </mark> folder, there are three contracts written in Solidity. We will be using these contracts for this tutorial. Also, remember to look at the <mark>audit/test/</mark> folder, as we will discuss it later.
+    
+2. You can do static analysis with Slither in two ways. You can either use it in the terminal or as an [extension](https://marketplace.visualstudio.com/items?itemName=trailofbits.slither-vscode) in vscode. We'll explain both options so it's easy to understand.
+    
+    **On your terminal run** `slither .`
+    
+    ```plaintext
+    slither .
+    ```
+    
+    You'll see some results in three groups: high (green), medium (yellow), and low (green) vulnerabilities. You can achieve the same outcome by using the slither extension in vscode. **Just click on the slither icon and then run it.**
```

Comment:
Can we include a screenshot here to show people what the extension results should look like?
Something like this for example.

![image](https://user-images.githubusercontent.com/56264430/224981540-fe910fd7-7802-4978-a1ae-89c6ead74e9f.png)